### PR TITLE
feat: add ccache module

### PR DESCRIPTION
**Pull request by enterJazz**
see https://ccache.dev/

@pogobanane wdyt

### CODE REVIEW

**Comment by enterJazz** [comment]
want to use this to cache kernel builds, not sure yet if it works ( i.e. if I need to do plumbing in buildLinux )

File: modules/ccache.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}:
+rec {
+  # https://nixos.wiki/wiki/CCache
+  programs.ccache = 
+  {
+    enable = true;
+    cacheDir = "/var/cache/ccache";
+  };
+
+  nixpkgs.overlays =
```
Comment (Mic92) [inline modules/ccache.nix:14]: You will need this package also in your flake.nix when importing nixpkgs.

You won't need an overlay, you can just override the package in a let statement.

**Comment by SuperSandro2000** [comment]
I tried using ccache myself and never got it to work even with very simple programs using cmake . Also it obviously doesn't work reliable if not all remote builders support it.

File: modules/ccache.nix (hunk 1 above)
Comment (enterJazz) [reply modules/ccache.nix:14]: will try this out and see if it works

**Comment by enterJazz** [comment]
> I tried using ccache myself and never got it to work even with very simple programs using cmake . Also it obviously doesn't work reliable if not all remote builders support it.

thanks for the input- will try Joerg's suggestion and let you know if it works for me

File: modules/ccache.nix (hunk 1 above)
Comment (enterJazz) [reply modules/ccache.nix]: @Mic92  like this?
https://github.com/Alper-Celik/SoundsOfGuis/blob/e06a940ee1810de0b60860788e19b1596e10dfe4/flake.nix#L18

**Comment by pogobanane** [comment]
I don't think we should deploy such caching to all hosts. But feel free to try it on vislor.

**Comment by enterJazz** [comment]
Things are not going too well w/ ccache w/ Kernel ; no one can really seem to get it to work
e.g. https://github.com/NixOS/nixpkgs/issues/153343#issuecomment-1379621036

will close this PR

**Pull request closed without merging**